torch/_inductor/scheduler.py

```diff
@@ -109,6 +109,43 @@ def fuse(node1: "BaseSchedulerNode", node2: "BaseSchedulerNode"):
     if node1.is_foreach() or node2.is_foreach():
         return ForeachKernelSchedulerNode.fuse(node1, node2)
     else:
+        _, (vars1, reduce1) = node1.group
+        _, (vars2, reduce2) = node2.group
+        if vars1 != vars2:
```

Comment:
Note that, for cuda kernels, group fn returns the product of `vars` not the tuple. You probably want `SchedulerNode.get_ranges()`.